### PR TITLE
MAGECLOUD-4919: Ability to run ece-tools on custom github source code

### PR DESCRIPTION
**Pull request by oshmyheliuk**
### Description
https://magento2.atlassian.net/browse/MAGECLOUD-4919

### Manual testing scenarios

1. Clone magento2ce https://github.com/magento/magento2ce
2. Add docker to composer.json
```
    "repositories": {
        "ece-tools": {
            "type": "git",
            "url": "git@github.com:magento\/ece-tools.git"
        },
        "magento-cloud-components": {
            "type": "git",
            "url": "git@github.com:magento\/magento-cloud-components.git"
        },
        "magento-cloud-docker": {
            "type": "git",
            "url": "git@github.com:magento\/magento-cloud-docker.git"
        },
        "magento-cloud-patches": {
            "type": "git",
            "url": "git@github.com:magento\/magento-cloud-patches.git"
        }
    },
    ...
    "require": {
         "magento/magento-cloud-docker": "dev-MAGECLOUD-4919 as 1.1.100",
        "magento/ece-tools": "dev-develop as 2002.1.100"
    }
```
     run composer update
3. Create the file `.magento.docker.yaml`

```yaml
# .magento.docker.yaml
services:
  php:
    version: "7.2"
    extensions:
      enabled:
        - xsl
  mysql:
    version: "10.0"
    # 10.4 for magento 2.4+
hooks:
  build: |
    set -e
    php ./vendor/bin/ece-tools run scenario/build/generate.xml
    php ./vendor/bin/ece-tools run scenario/build/transfer.xml
  deploy: |
    php ./vendor/bin/ece-tools run scenario/deploy.xml
  post_deploy: |
    php ./vendor/bin/ece-tools run scenario/post-deploy.xml
mounts:
  var:
    path: "var"
  app-etc:
    path: "app/etc"
  pub-media:
    path: "pub/media"
  pub-static:
    path: "pub/static"
```

4. Build a PHP image:

```bash
 docker build -t test/php:7.2-cli ../magento-cloud-docker/images/php/7.2-cli
```

5. Create the extending file `docker-compose.override.yml`:

```yaml
version: '2.1'
services:
  web:
    volumes:
      - './pub/index.php:/app/pub/index.php:ro'
  build:
    image: test/php:7.2-cli
  deploy:
    image: test/php:7.2-cli
```

6. Create .magento.env.yaml file
```
stage:
  global:
    DEPLOYED_MAGENTO_VERSION_FROM_GIT: 2.3.3
```
7. Run build compose command with installation type option
```
php vendor/bin/ece-docker build:compose --mode=production --installation-type=git
```
8. Install magento on docker as in reqular installation flow
9. Test developer and producion modes 

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

File: src/Config/Source/SourceInterface.php [hunk 1]
```diff
@@ -87,7 +87,7 @@ interface SourceInterface
     public const PHP_ENABLED = self::PHP . '.enabled';
     public const PHP_ENABLED_EXTENSIONS = self::PHP . '.extensions.enabled';
     public const PHP_DISABLED_EXTENSIONS = self::PHP . '.extensions.disabled';
-
+    public const INSTALLATION_TYPE = 'install.type';
```
Comment (shiftedreality) [inline src/Config/Source/SourceInterface.php:91]: Looks like the new line is missed

File: src/Command/BuildCompose.php [hunk 2]
```diff
@@ -205,6 +205,13 @@ protected function configure(): void
                 InputOption::VALUE_OPTIONAL,
                 'Cloud environment variables'
             )
+            ->addOption(
+                Source\CliSource::OPTION_INSTALLATION_TYPE,
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'Sets that magento installed from git',
```
Comment (shiftedreality) [inline src/Command/BuildCompose.php]: magento must be uppercase, please also rephrase this, as it's confusing

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: src/Command/BuildCompose.php (hunk 2 above)
Comment (oshmyheliuk) [reply src/Command/BuildCompose.php]: Yeah, it was a description of the previous implementation. Thanks, I'll fix

Comment (oshmyheliuk) [reply src/Command/BuildCompose.php]: updated

File: src/Config/Source/SourceInterface.php (hunk 1 above)
Comment (oshmyheliuk) [reply src/Config/Source/SourceInterface.php:91]: added

**Review by shiftedreality — APPROVED** [review]

**Review by BaDos — APPROVED** [review]

**Comment by BaDos** [comment]
@oshmyheliuk Plz cover by FT

**Comment by oshmyheliuk** [comment]
will be covered in this task https://magento2.atlassian.net/browse/MAGECLOUD-5592

**Comment by mveeramneni** [comment]
QA Approved

**Comment by mveeramneni** [comment]
QA Approved

**Pull request merged**